contracts/test/v0.8/functions/v1/FunctionsClient.test.ts

```diff
@@ -173,3 +173,51 @@ describe('Functions Client', () => {
     })
   })
 })
+
+describe('Faulty Functions Client', () => {
```

Comment:
Does this test fail without the change to Router made above? If so, why?

Reply:
Yes.
This is because client contract's callback being empty uses less gas than  `gasForCallExactCheck`.
`Router._callback`'s `gasUsed` (previously set to `g` minus `gas()` after callback) underflows because `g` (gas() - gasForCallExactCheck) is less `gas()` (when taken after the call)

Reply:
Ah, interesting, thanks for explaining! Mind adding a comment to the test, please?

Reply:
A comment to the test explaining something is no longer in the code?

Reply:
Maybe a comment to the yule code would be better suited

Reply:
I was thinking about something along the lines of "make sure we charge zero for empty callback and not add any overhead" but the existing "can complete requests with an empty callback" is good enough.